File: 8kyu/Python/calculateAverage.py
'''
Write a function which calculates the average of the numbers in a given list.

Note: Empty arrays should return 0.
'''

# method 1
def find_average(nums):
  # return average by summing the nums array, dividing by length of array if the array has at least 1 element in it, else return zero
    return sum(nums)/len(nums) if len(nums) > 0 else 0

# method 2: using lambda
find_average = lambda nums : sum(nums)/len(nums) if len(nums) > 0 else 0

File: 8kyu/JavaScript/calculateAverage.js
/*
Write a function which calculates the average of the numbers in a given list.

Note: Empty arrays should return 0.
*/

// method 1
const findAverage = arr => {
  // get sum by reducing array
  let sum = arr.reduce((a, b) => a + b, 0);
  // divide sum by the length of the array
  let average = +(sum / arr.length);
  // return that average or 0 if its falsy
  return average || 0;
};

// method 2: same as method 1 but putting all on one line, not using variables. more concise, less readable
const findAverage = arr => +((arr.reduce((a, b) => a + b, 0))/arr.length) || 0;
